explorer/utils/azureUtils.ts

```diff
@@ -15,3 +50,127 @@ export function browseAzurePortal(context?: AzureRegistryNode | AzureRepositoryN
+ */
+export async function getAzureRepositories(registry: Registry): Promise<Repository[]> {
+    const allRepos: Repository[] = [];
+    let repo: Repository;
+    let resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+    let subscriptionId = registry.id.slice('/subscriptions/'.length, registry.id.search('/resourceGroups/'));
+    const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+    //get the actual subscription object by using the id found on the registry id above
+    const subscription = subs.find(function (sub): boolean {
+        return sub.subscriptionId === subscriptionId;
+    });
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount();
+    if (!azureAccount) {
+        return [];
+    }
+
+    const { accessToken, refreshToken } = await getTokens(registry);
+
+    if (accessToken && refreshToken) {
+
+        await request.get('https://' + registry.loginServer + '/v2/_catalog', {
+            auth: {
+                bearer: accessToken
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                const repositories = JSON.parse(body).repositories;
+                for (let i = 0; i < repositories.length; i++) {
+                    repo = new Repository(azureAccount, registry, repositories[i], subscription, resourceGroup, accessToken, refreshToken);
+                    allRepos.push(repo);
+                }
+            }
+        });
+    }
+    //Note these are ordered by default in alphabetical order
+    return allRepos;
+}
+
+/**
+ * this function gets the refresh and access tokens for a given registry
+ * @param registry : the registry to get credentials for
+ * @returns : the updated refresh and access tokens which can be used to generate a header for an API call
+ */
+export async function getTokens(registry: Registry): Promise<{ refreshToken: any, accessToken: any }> {
+    let subscriptionId = registry.id.slice('/subscriptions/'.length, registry.id.search('/resourceGroups/'));
+    const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+    //get the actual subscription object by using the id found on the registry id above
+    const subscription = subs.find(function (sub): boolean {
+        return sub.subscriptionId === subscriptionId;
+    });
+    const tenantId: string = subscription.tenantId;
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount()
+    if (!azureAccount) {
+        return;
+    }
+
+    const session: AzureSession = azureAccount.sessions.find((s, i, array) => s.tenantId.toLowerCase() === tenantId.toLowerCase());
+    const { accessToken, refreshToken } = await acquireToken(session);
+
+    //regenerates in case they have expired
+    if (accessToken && refreshToken) {
+        let refreshTokenARC;
```

Comment:
>ARC [](start = 24, length = 3)

acr?

Reply:
not entirely sure what ARC refers to, but this was the convention used in other files such as azureRegistryNodes.ts